File: code/dobot_quad_sdk-main/high_level/cpp/e10_config_demo.cpp
// Example 10: Speed Ratio & Obstacle Avoidance configuration demo.
// Usage: ./e10_config_demo [server_address]
//
// Demonstrates:
//   - Querying and setting speed ratio (10-100)
//   - Toggling obstacle avoidance on/off
//   - Walking at different speed ratios to see the difference

#include "robot_client.h"

int main(int argc, char** argv)
{
    robot::Client client(argc > 1 ? argv[1] : "192.168.5.2:50051");
    robot::enable_safety_ready(client);

    // --- Speed Ratio Demo ---
    std::cout << std::string(50, '=') << std::endl;
    std::cout << "Speed Ratio Demo" << std::endl;
    std::cout << std::string(50, '=') << std::endl;

    // Query current speed ratio
    int current_sr = client.get_speed_ratio();
    std::cout << "Current speed ratio: " << current_sr << "\n" << std::endl;

    // Set base speed ratio to 10 (slow)
    std::cout << "--- Walk forward 1m at speed_ratio=10 (slow) ---" << std::endl;
    client.set_speed_ratio(10);
    current_sr = client.get_speed_ratio();
    std::cout << "Current speed ratio: " << current_sr << "\n" << std::endl;
    client.walk_forward(1.0);
    std::this_thread::sleep_for(std::chrono::seconds(1));

    // Walk forward with explicit speed_ratio=100 (temporary override, restores after)
    client.walk_forward(1.0, 100);
    current_sr = client.get_speed_ratio();
    std::cout << "Current speed ratio: " << current_sr << "\n" << std::endl;

    // --- Obstacle Avoidance Demo ---
    std::cout << "\n" << std::string(50, '=') << std::endl;
    std::cout << "Obstacle Avoidance Demo" << std::endl;
    std::cout << std::string(50, '=') << std::endl;

    // Disable obstacle avoidance
    std::cout << "\n--- Enabling obstacle avoidance ---" << std::endl;
    auto oa_resp = client.set_obstacle_avoidance("off");
    std::cout << "  OA enabled: " << (oa_resp.current_enabled() ? "true" : "false") << std::endl;
    std::cout << "  OA get: " << (client.get_obstacle_avoidance() ? "true" : "false") << std::endl;

    // Enable obstacle avoidance
    oa_resp = client.set_obstacle_avoidance(true);
    std::cout << "  OA enabled: " << (oa_resp.current_enabled() ? "true" : "false") << std::endl;
    std::cout << "  OA get: " << (client.get_obstacle_avoidance() ? "true" : "false") << std::endl;

    std::cout << "\nDemo complete." << std::endl;
    return 0;
}

File: code/dobot_quad_sdk-main/high_level/python/examples/e10_config_demo.py
#!/usr/bin/env python3
"""Example 10: Speed Ratio & Obstacle Avoidance configuration demo.
Usage: python e10_config_demo.py [server_address]

Demonstrates:
  - Querying and setting speed ratio (10-100)
  - Toggling obstacle avoidance on/off
  - Walking at different speed ratios to see the difference
"""
import sys
import time
from dobot_quad import RobotClient


def main():
    robot = RobotClient(sys.argv[1] if len(sys.argv) >
                        1 else "192.168.5.2:50051")
    robot.enable_safety_ready()

    # --- Speed Ratio Demo ---
    print("=" * 50)
    print("Speed Ratio Demo")
    print("=" * 50)

    # Query current speed ratio
    current_sr = robot.get_speed_ratio()
    print(f"Current speed ratio: {current_sr}\n")

    # Walk forward at low speed (ratio=30)
    print("--- Walk forward 2m at speed_ratio=10 (slow) ---")
    robot.set_speed_ratio(10)
    # Query current speed ratio
    current_sr = robot.get_speed_ratio()
    print(f"Current speed ratio: {current_sr}\n")
    robot.walk_forward(1.0)
    time.sleep(1)
    # Query current speed ratio
    robot.walk_forward(1.0, speed_ratio=100)
    current_sr = robot.get_speed_ratio()
    print(f"Current speed ratio: {current_sr}\n")

    # --- Obstacle Avoidance Demo ---
    print("\n" + "=" * 50)
    print("Obstacle Avoidance Demo")
    print("=" * 50)

    # Enable obstacle avoidance
    print("\n--- Enabling obstacle avoidance ---")
    resp = robot.set_obstacle_avoidance("off")
    print(f"  OA enabled: {resp.current_enabled}")
    print(f"  OA get: {robot.get_obstacle_avoidance()}")
    resp = robot.set_obstacle_avoidance(True)
    print(f"  OA enabled: {resp.current_enabled}")
    print(f"  OA get: {robot.get_obstacle_avoidance()}")

    # robot.stand_down()
    print("\nDemo complete.")
    return 0


if __name__ == "__main__":
    sys.exit(main())
